Sources/CartonHelpers/DiagnosticsParser.swift

```diff
@@ -0,0 +1,211 @@
+    var lineIdx = 0
+    
+    var diagnostics = [String.SubSequence:[CustomDiagnostic]]()
+    
+    var currFile: String.SubSequence?
+    var fileMessages = [CustomDiagnostic]()
+    
+    while lineIdx < lines.count {
+      let line = lines[lineIdx]
+      if let file = line.matches(regex: Regex.enterFile) {
+        if let currFile = currFile {
+          diagnostics[currFile] = fileMessages
+        }
+        currFile = file
+        fileMessages = []
+      } else if currFile != nil {
+        if let message = line.matches(regex: Regex.line) {
+          let components = message.split(separator: ":")
+          if components.count > 3 {
+            lineIdx += 1
+            fileMessages.append(
+              .init(
+                kind: CustomDiagnostic.Kind(rawValue: String(components[2].trimmingCharacters(in: .whitespaces))) ?? .note,
+                line: components[0],
+                char: components[1],
+                code: String(lines[lineIdx]),
+                message: components[3]
+              )
+            )
+          }
+        }
+      } else {
+        terminal.write(String(line) + "\n", inColor: .cyan)
+      }
+      lineIdx += 1
+    }
+    if let currFile = currFile {
+      diagnostics[currFile] = fileMessages
+    }
+    
+    for (file, messages) in diagnostics.sorted(by: { $0.key < $1.key }) {
+      guard messages.count > 0 else { continue }
+      terminal.write("\(" \(file) ", color: "[1m", "[7m")\n\n") // bold, reversed
+      // Group messages that occur on sequential lines to provie a more readable output
+      var groupedMessages = [[CustomDiagnostic]]()
+      for message in messages {
+        if let lastLineStr = groupedMessages.last?.last?.line,
+           let lastLine = Int(lastLineStr),
+           let line = Int(message.line),
+           lastLine == line - 1 || lastLine == line {
+          groupedMessages[groupedMessages.count - 1].append(message)
+        } else {
+          groupedMessages.append([message])
+        }
+      }
+      for messages in groupedMessages {
+        // Output the diagnostic message
+        for message in messages {
+          terminal.write("  \(" \(message.kind.rawValue.uppercased()) ", color: message.kind.color, "[37;1m") \(message.message)\n") // 37;1: bright white
+        }
+        let maxLine = messages.map(\.line.count).max() ?? 0
+        for (offset, message) in messages.enumerated() {
+          func flush() {
+            // Get all diagnostics for a particular line.
+            let allChars = messages.filter { $0.line == message.line }.map(\.char)
+            // Output the code for this line, syntax highlighted
+            terminal.write("  \("\(message.line.padding(toLength: maxLine, withPad: " ", startingAt: 0)) | ", color: "[36m")\(Self.highlighter.highlight(message.code))\n") // 36: cyan
+            terminal.write("  " + "".padding(toLength: maxLine, withPad: " ", startingAt: 0) + " | ", inColor: .cyan)
```

Comment:
<!--
  1 failure:  Line should be 10...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: Sources/CartonHelpers/DiagnosticsParser.swift;
  Line: 179;
-->
- :no_entry_sign: Line should be 100 characters or less: currently 117 characters (`line_length`)